### ISSUE TITLE
Remixed Bundles Advanced Game Option

### ISSUE THREAD

**Issue by kihashi** [kihashi/stardew_community_checklist#163]
![image](https://user-images.githubusercontent.com/610242/102734226-5125ce00-430d-11eb-9599-651c3a1b2b2e.png)

Seems like CA is adding the ability to randomize the community center bundles in 1.5. 

How should we handle this? Depending on how it is implemented in the game, we might need to be able to handle a seed value or parse the save file to set things up. 

It's also likely that we'll need to add all existing items to support this feature.

**Comment by kihashi** [comment]
https://stardewvalleywiki.com/Bundles#Remixed_Bundles

So it looks like there are 2 variations-

In the first case, some items are changed to be other items.
In the second case, some bundles are changed to other bundles.

AFAICT, it does not appear to be completely random based on that. 

So, how do we deal with that? Do we add a "remix bundle designer" that includes all of the variations and allows the user to swap out based on what they see in their game?

**Comment by mihmilicio** [comment]
I think that's the way... Do you have a layout in mind to it? I'd say something like this would be quite intuitive: 

![image](https://user-images.githubusercontent.com/37940750/103469362-76213480-4d42-11eb-9483-62da401fcc4b.png)
(don't mind the details, just the general idea 😅)

What I'm having trouble imagining is where the "remix bundle designer" would fit into the app...

**Comment by kihashi** [comment]
I'm still thinking about how to design it. The bundle substitutions seem to be the most straight forward since you either have one or the other.

Keep in mind, though, that we can't use any game assets.

**Comment by kihashi** [comment]
I added a splash screen to let people know that we're working on it- #167

**Comment by mihmilicio** [comment]
> 
> I'm still thinking about how to design it. The bundle substitutions seem to be the most straight forward since you either have one or the other. Keep in mind, though, that we can't use any game assets.

Yeah, I read about the bundle substitutions, but if the design is too specific it tends to be odd and confusing, I think. IMO the ideal is to show all bundles, and those that are always present just have their checkboxes (or whatever component that will show state) marked and disabled. This way they would serve as an example of the logic and, at the same time, highlight the changeable bundles that the user should pay attention.

About the game assets, are you sure? I've seen other tools using it 😅. I also asked in the forum, just to be sure.

**Comment by kihashi** [comment]
> About the game assets, are you sure? I've seen other tools using it 😅. I also asked in the forum, just to be sure.

I have not thoroughly read the terms of service of the game, but generally speaking, game assets are copyrighted. You would need special permission or some sort of open license to use it. We can use text from the wiki because it is licensed under the Creative Commons license. The images on the wiki from the game, however, appear to be specially licensed for use on the Wiki.

![image](https://user-images.githubusercontent.com/610242/103492154-f73a0380-4df6-11eb-9031-599231d6025c.png)

As to other tools using game assets, I can only assume that they are either relying on the benevolence of ConcernedApe or have asked for special permission.

**Comment by barnwellrd** [comment]
I've started looking into this from a save data import point of view. I'm not 100% certain, but it looks like some of the bundles can be randomized. From the links above and what I've seen in game, there are two possibilities for remixed bundles. One is a brand new bundle, and the other is where an existing bundle's items are randomized (maybe from a preset list).

The solution I am working on will let a user import their save file by using a "Remixed Bundle" toggle on the settings page, and then the bundleData is parsed out and matched to a master item list to compile a remixed bundle list for use in the UI.

I haven't started on the UI portion yet, just the data processing, but the implementation of a master item list will be pretty big project to include the skills, seasons, and source if it has to be comprehensive, unless there is a preset list of items that is smaller that the remixed bundles pull from.

Here are some pastebins of the data I've already gotten parsed out:

- bundleData in XML from the save file: https://pastebin.com/mHt2i1SE
- Parsed Bundle Data as JSON (WIP but matches a combo of the bundles.json and items.json data): https://pastebin.com/9FC7fhSg
- Master Item List JSON (should match the existing items.json layout): https://pastebin.com/M2esAxe4

**Comment by mihmilicio** [comment]
We have to keep in mind that we shouldn't have the "upload save" as the only option, because there's players from other platforms and consoles.

Also, I got some [answers in the forum](https://forums.stardewvalley.net/threads/is-it-allowed-to-use-game-assets-for-helpers.4107/#post-24981) about the assets, and, personally, I'd use it. But it's not my call.
